provider/online.go

```diff
@@ -34,6 +33,19 @@ type OnlineStoreTable interface {
 	Get(entity string) (interface{}, error)
 }
 
+type VectorStore interface {
+	CreateIndex(feature, variant, entity string, vectorType VectorType) (VectorStoreTable, error)
+	GetIndex(feature, variant string) (string, error) // TODO: determine the type of index
```

Comment:
TODO?

Reply:
I was thinking about this some more and had another fun idea:

Let make all VectorStores OnlineStores,

So it becomes:
type VectorStore interface {
  OnlineStore
  GetIndex
  DeleteIndex
  }
  
  Note that Provider and Close should be removed since you'll get those via OnlineStore's interface

Reply:
Doing this won't change any other code, it's so that every VectorStore implements GetTable which makes life easier. We can talk through it if its still confusing